test/manifold/burnredeem721.js

```diff
@@ -0,0 +1,516 @@
+      let end = start + 300;
+      let burnRedeemData = web3.eth.abi.encodeParameters(["address", "uint256", "uint256", "bytes32[]"], [creator.address, 1, 0, []]);
+
+      // Mint burnable tokens
+      await burnable721.mintBase(anyone1, { from: owner });
+      await burnable721_2.mintBase(anyone1, { from: owner });
+
+      // Ensure that the creator contract state is what we expect before mints
+      let balance = await creator.balanceOf(anyone1);
+      assert.equal(0, balance);
+      balance = await burnable721.balanceOf(anyone1);
+      assert.equal(1, balance);
+      
+      await burnRedeem.initializeBurnRedeem(
+        creator.address,
+        1,
+        {
+          startDate: start,
+          endDate: end,
+          totalSupply: 10,
+          identical: false,
+          storageProtocol: 1,
+          location: "XXX",
+          cost: 0,
+          burnSet: [
+            {
+              requiredCount: 1,
+              items: [
+                {
+                  validationType: 1,
+                  contractAddress: burnable721.address,
+                  minTokenId: 0,
+                  maxTokenId: 0,
+                  merkleRoot: ethers.utils.formatBytes32String("")
+                }
+              ]
+            }
+          ],
+        },
+        {from:owner}
+      );
+
+      // Reverts due to wrong contract
+      await truffleAssert.reverts(burnable721_2.methods['safeTransferFrom(address,address,uint256,bytes)'](anyone1, burnRedeem.address, 1, burnRedeemData, {from:anyone1}), "Invalid burn token");
+
+      // Passes with right token id
+      await truffleAssert.passes(burnable721.methods['safeTransferFrom(address,address,uint256,bytes)'](anyone1, burnRedeem.address, 1, burnRedeemData, {from:anyone1}));
+
+      // Ensure tokens are burned/minted
+      balance = await burnable721.balanceOf(anyone1);
+      assert.equal(0, balance);
+      balance = await creator.balanceOf(anyone1);
+      assert.equal(1, balance);
+    });
+
+    it('burnRedeem test', async function() {
+
+      // Test initializing a new burn redeem
+      let start = (await web3.eth.getBlock('latest')).timestamp-30; // seconds since unix epoch
+      let end = start + 300;
+
+      // Mint burnable tokens
+      for (let i = 0; i < 3; i++) {
+        await burnable721.mintBase(anyone1, { from: owner });
+        await burnable721_2.mintBase(anyone1, { from: owner });
+      }
+
+      // Ensure that the creator contract state is what we expect before mints
+      let balance = await creator.balanceOf(anyone1);
+      assert.equal(0, balance);
+      balance = await burnable721.balanceOf(anyone1);
+      assert.equal(3, balance);
+      balance = await burnable721_2.balanceOf(anyone1);
+      assert.equal(3, balance);
+
+      const merkleElements = [];
+      merkleElements.push(ethers.utils.solidityPack(['uint256'], [3]));
+      merkleElements.push(ethers.utils.solidityPack(['uint256'], [10]));
```

Comment:
can you try this for me to make sure it's the same as keccak256 is a node-only pacakge:
```
import { keccak256 } from '@ethersproject/solidity';
const leaves = tokenIds.map(
      (x: string) => keccak256(['uint256'], [x])
    );
);
```

I want to ensure the behavior is the same.

Reply:
or actually you could also do this, maybe that's more right:

```
import { keccak256, pack } from '@ethersproject/solidity';
merkleElements.push(pack(['uint256'], [15]));
...
merkleTreeWithValues = new MerkleTree(merkleElements, keccak256, { hashLeaves: true, sortPairs: true });
```

Reply:
So far the only version I can get working on the FE is:

```
    const leaves = tokenIds.map(
      (x: string) => keccak256(['uint256'], [parseInt(x)])
    );
    const tree = new MerkleTree(leaves);
    const root = tree.getRoot().toString('hex');
    ```

Reply:
ugh nevermind it hates ethers version of the method because it requires two params.... blehhhhh what do i do?

Reply:
finally got it, shimmed Buffer 👍